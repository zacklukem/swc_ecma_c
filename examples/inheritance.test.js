function Animal(name) {
  console.log("Animal constructor called with:", name);
  this.name = name;
}

Animal.prototype.sayName = function () {
  console.log("My name is", this.name);
};

function Dog(name, breed) {
  Animal.call(this, name);
  console.log("Dog constructor called with:", name, breed);
  this.breed = breed;
}

Dog.prototype = Object.create(Animal.prototype);

Dog.prototype.sayFullName = function () {
  console.log("My name is", this.name, "and I am a", this.breed);
};

let d = new Dog("Fido", "Labrador");
d.sayFullName();
d.sayName();

// TODO: instanceof test
