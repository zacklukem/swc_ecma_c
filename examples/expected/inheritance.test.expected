Animal constructor called with: Fido
Dog constructor called with: Fido Labrador
My name is Fido and I am a Labrador
My name is Fido
